pkg/manifests/nginx.go

```diff
@@ -111,10 +113,11 @@ var (
 
 // NginxIngressConfig defines configuration options for required resources for an Ingress
 type NginxIngressConfig struct {
-	ControllerClass string         // controller class which is equivalent to controller field of IngressClass
-	ResourceName    string         // name given to all resources
-	IcName          string         // IngressClass name
-	ServiceConfig   *ServiceConfig // service config that specifies details about the LB, defaults if nil
+	ControllerClass       string                          // controller class which is equivalent to controller field of IngressClass
+	ResourceName          string                          // name given to all resources
+	IcName                string                          // IngressClass name
+	ServiceConfig         *ServiceConfig                  // service config that specifies details about the LB, defaults if nil
+	DefaultSSLCertificate *v1alpha1.DefaultSSLCertificate // namespace/name used to create SSL certificate for the default HTTPS server (catch-all)
```

Comment:
this should be a namespace / name instead of the crd api. This will always boil down to a namespace name. Doesn't make sense to mix the application logic across the entire codebase instead of using different abstractions.

By just using namespace/name we make it easier to write this part and more clear what the NginxIngressConfig can be used for